File: library/src/main/java/com/getstream/sdk/chat/storage/converter/ExtraDataConverter.java
package com.getstream.sdk.chat.storage.converter;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.HashMap;

public class ExtraDataConverter {
    static Gson gson = new Gson();

    @TypeConverter
    public static HashMap<String, Object> stringToMap(String data) {
        if (data == null) {
            return new HashMap<>();
        }

        Type mapType = new TypeToken<HashMap<String, Object>>() {
        }.getType();

        return gson.fromJson(data, mapType);
    }

    @TypeConverter
    public static String mapToString(HashMap<String, Object> someObjects) {
        return gson.toJson(someObjects);
    }
}

File: app/src/main/java/io/getstream/chat/android/core/poc/library/api/ExtraDataConverter.kt
package io.getstream.chat.android.core.poc.library.api

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import io.getstream.chat.android.core.poc.library.json.ChatGson
import java.lang.reflect.Type


object ExtraDataConverter {
    var gson = ChatGson.instance
    @TypeConverter
    fun stringToMap(data: String?): HashMap<String, Any> {
        if (data == null) {
            return HashMap()
        }
        val mapType: Type = object : TypeToken<HashMap<String?, Any?>?>() {}.type
        return gson.fromJson(data, mapType)
    }

    @TypeConverter
    fun mapToString(someObjects: HashMap<String?, Any?>?): String {
        return gson.toJson(someObjects)
    }
}
